### ISSUE TITLE
Incompatible return value type generated by monkeytype

### ISSUE THREAD

**Issue by aabramovrepo** [Instagram/MonkeyType#187]
Dear All,

running mypy on the code:

```
from typing import Dict, List, Union

ListItemType = List[Dict[str, Union[str, float, int]]]


def get_item_list() -> ListItemType:
    my_list = list()

    item_1 = {'param_1': 'string one',
              'param_2': 3.4,
              'param_3': 10}

    item_2 = {'param_1': 'string two',
              'param_2': 55.4,
              'param_3': 22}

    my_list.append(item_1)
    my_list.append(item_2)

    return my_list


def main():
    my_list = get_item_list()


if __name__ == '__main__':
    main()

```

I get the following error:

`error: Incompatible return value type (got "List[Dict[str, object]]", expected "List[Dict[str, Union[str, float, int]]]")
`

the annotations have been generated by `monkeytype`, any ideas or suggestions? Can `monkeytype` handle list invariance?

Best,
Alexey

**Comment by carljm** [comment]
MonkeyType is not a type checker, so it knows nothing about list invariance. It just observes runtime types, constructs the most exact type it can from them, and applies that type. It is just giving you information about your actual runtime types, which you can do with as you will. It has no idea what type a typechecker might infer from your code, or whether the type-checker will consider that type compatible with your runtime types.

In other words, MonkeyType is behaving as expected here and there is really nothing better it could do. It is up to you to resolve mypy's unhappiness however you prefer, whether that is changing the annotation to `List[Dict[str, object]]` or providing an explicit variable annotation within the function to prevent mypy from inferring `List[Dict[str, object]]`.

**Comment by aabramovrepo** [comment]
@carljm many thanks for a such a detailed response, very helpful!
